src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -242,10 +267,40 @@ public void Tick(DateTimeOffset now)
 
                 if (_timeoutAction == TimeoutAction.SendTimeoutResponse)
                 {
+                    SetTimeoutResponse();
+                }
+
+                Timeout();
+            }
+            else if (_timingReads)
+            {
+                _readTimingElapsed += timestamp - _lastTimestamp;
+
+                if (_readTimingElapsed > _frame.RequestBodyTimeout.Ticks)
```

Comment:
This along with RequestHeadersTimeout gives the developer a simple/convenient way to set an absolute timeout for a client to send a complete request. Even if it's possible to achieve the same thing by doing some math and selecting the appropriate min rate and max size, that seems more complicated than necessary for many scenarios.